### PR TITLE
fix(jupyter): more robust Deno.jupyter namespace

### PR DESCRIPTION
**Pull request by bartlomieju**
Blocked by https://github.com/denoland/deno/pull/20704

### CODE REVIEW

File: cli/tests/testdata/jupyter/integration_test.ipynb [hunk 1]
```diff
@@ -665,19 +667,24 @@
...
-     "evalue": "Cannot read properties of undefined (reading 'broadcast')",
-     "output_type": "error",
-     "traceback": [
-      "Stack trace:",
-      "TypeError: Cannot read properties of undefined (reading 'broadcast')",
-      "    at <anonymous>:2:20"
-     ]
+     "data": {
+      "text/html": [
+       "<b>Complete ✅</b>"
```
Comment (rgbkrk) [inline cli/tests/testdata/jupyter/integration_test.ipynb:677]: 🎉

File: cli/tests/testdata/jupyter/integration_test.ipynb [hunk 2]
```diff
@@ -665,19 +667,24 @@
...
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {},
+     "execution_count": 21,
+     "metadata": {},
+     "output_type": "execute_result"
     }
```
Comment (rgbkrk) [inline cli/tests/testdata/jupyter/integration_test.ipynb:688]: For a separate PR, we should make sure to not send an execute_result if we don't have any payload.

Comment (bartlomieju) [reply cli/tests/testdata/jupyter/integration_test.ipynb:688]: Hm, that looks like a bug, we should be skipping this message if the only value is `undefined`.

File: cli/tools/jupyter/server.rs [hunk 3]
```diff
@@ -391,15 +391,18 @@ impl JupyterServer {
...
+      if !output.is_empty() {
+        msg
+          .new_message("execute_result")
+          .with_content(json!({
+              "execution_count": self.execution_count,
+              "data": output,
+              "metadata": {},
+          }))
+          .send(&mut *self.iopub_socket.lock().await)
+          .await?;
```
Comment (rgbkrk) [inline cli/tools/jupyter/server.rs:404]: 👍🏻

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
Confirmed that `Deno.jupyter.broadcast` is working well via this branch (fixes `main` not having access to `Deno.jupyter`). 

image

**Review by rgbkrk — APPROVED** [review]

**Review by mmastrac — APPROVED** [review]
LGTM

**Pull request merged**